### PR TITLE
Raw Server Errors in Artifacts

### LINKED ISSUE #85: [Feature] Put raw server errors in artifacts
For any error that's going to be reported as part of a `[ServerErrors]` bug, there should be a corresponding AzDO pipeline artifact containing the unsanitized text of the error.

### PR DESCRIPTION
**Pull request by navya9singh**
Fixes #85

### CODE REVIEW

**Comment by amcasey** [comment]
I think you need to put "Fixes #85" for GH to notice and close the issue.

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:65]: Extra space?

File: src/main.ts
Comment (amcasey) [inline src/main.ts:307]: "raw error text"

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:19]: Please update the usage text as well.

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:84]: This isn't relevant for the raw error file - it's for configuring a repro run.

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:411]: I'm missing something.  This appears to be writing the request string to the raw errors file, but the sample run you shared had the call stack from the response.  Is there some reason this won't be exactly the same text as the replay script?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: src/utils/exerciseServer.ts
Comment (navya9singh) [reply src/utils/exerciseServer.ts]: You're right. I added this part initially to check if raw error file was being created and yes now the file has call stack from the response. I've removed it now.

File: src/utils/exerciseServer.ts
Comment (navya9singh) [reply src/utils/exerciseServer.ts]: I've added `<requests_path>` for raw error too, like replay script

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:18]: As we discussed offline, I think you can revert this whole file.

File: src/main.ts
Comment (amcasey) [inline src/main.ts:258]: Can you please add a comment with the parameter name, like the line above?  We like to do that for literal arguments.

File: src/main.ts
Comment (amcasey) [inline src/main.ts:653]: I think we can just write it to its final location.

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:41]: Maybe just revert the whole file?  `git checkout -f main -- src/utils/exerciseServer.ts` should work.

File: src/main.ts
Comment (amcasey) [inline src/main.ts:258]: Can you please confirm that this is supposed to end up in `rawErrorsScriptPath` and not `path.join(resultDirPath, rawErrorsScriptFileName)` (i.e. where used to copy it to below)?

File: src/main.ts
Comment (amcasey) [inline src/main.ts:215]: This should no longer be necessary (and will probably cause it to exit and print the usage message).

File: src/main.ts
Comment (amcasey) [inline src/main.ts:189]: I'm sure this was my own suggestion, but I think it might be clearer to replace "rawErrorsScript" with just "rawError" (no 's') everywhere, since there's only one error and it's not a script.

File: src/main.ts
Comment (navya9singh) [reply src/main.ts]: I ran the tests and raw error file is not generated if it ends up at `rawErrorsScriptPath`. So I changed it back to end up at where it was copied over (`path.join(resultDirPath, rawErrorsScriptFileName)` ). It required parameters like `resultDirPath` that have later been defined in file, so I cannot make the file write to the final destination directly here.

File: src/main.ts
Comment (amcasey) [inline src/main.ts:189]: Sorry if my previous request was unclear, but can you please change "rawErrorScript" to "rawError"?

File: src/main.ts
Comment (amcasey) [reply src/main.ts]: Sorry, the diff made it hard to tell that the copy was in the caller.  This is fine.

File: src/main.ts
Comment (amcasey) [reply src/main.ts]: We could, but it would require unnecessary gymnastics and this is consistent with how we handle the replay script.

File: src/main.ts
Comment (amcasey) [inline src/main.ts:307]: "in the"

File: src/utils/exerciseServer.ts
Comment (amcasey) [inline src/utils/exerciseServer.ts:63]: Obviously, don't merge this. :wink:

**Review by amcasey — APPROVED** [review]
LGTM after some minor cleanup.

File: src/main.ts
Comment (navya9singh) [reply src/main.ts]: Done :)

**Comment by amcasey** [comment]
🎉

**Pull request merged**
